platform/files/analyzed.go

```diff
@@ -0,0 +1,228 @@
+}
+
+// NOTE: This struct MUST be kept in sync with `LayersMetadata`.
+// It exists for situations where the `App` field type cannot be
+// guaranteed, yet the original struct data must be maintained.
+type LayersMetadataCompat struct {
+	App          interface{}                `json:"app" toml:"app"`
+	BOM          *LayerMetadata             `json:"sbom,omitempty" toml:"sbom,omitempty"`
+	Buildpacks   []buildpack.LayersMetadata `json:"buildpacks" toml:"buildpacks"`
+	Config       LayerMetadata              `json:"config" toml:"config"`
+	Launcher     LayerMetadata              `json:"launcher" toml:"launcher"`
+	ProcessTypes LayerMetadata              `json:"process-types" toml:"process-types"`
+	RunImage     RunImageForRebase          `json:"runImage" toml:"run-image"`
+	Stack        Stack                      `json:"stack" toml:"stack"`
+}
+
+func (m *LayersMetadata) LayersMetadataFor(id string) buildpack.LayersMetadata {
+	for _, bpMD := range m.Buildpacks {
+		if bpMD.ID == id {
+			return bpMD
+		}
+	}
+	return buildpack.LayersMetadata{}
+}
+
+type LayerMetadata struct {
+	SHA string `json:"sha" toml:"sha"`
+}
+
+type RunImageForRebase struct {
+	TopLayer  string `json:"topLayer" toml:"top-layer"`
+	Reference string `json:"reference" toml:"reference"`
+
+	// added in Platform 0.12
+	Image   string   `toml:"image,omitempty" json:"image,omitempty"`
+	Mirrors []string `toml:"mirrors,omitempty" json:"mirrors,omitempty"`
+}
+
+type RunImage struct {
+	Reference string `toml:"reference"`
+	// Image specifies the repository name for the image.
+	// When exporting to a daemon, the restorer uses this field to pull the run image if needed for the extender;
+	// it can't use reference because this may be a daemon image ID if analyzed.toml was last written by the analyzer.
+	Image string `toml:"image,omitempty"`
+	// Extend if true indicates that the run image should be extended by the extender.
+	Extend         bool            `toml:"extend,omitempty"`
+	TargetMetadata *TargetMetadata `json:"target,omitempty" toml:"target,omitempty"`
+}
+
+type TargetMetadata struct {
+	ID          string `json:"id" toml:"id"`
+	OS          string `json:"os" toml:"os"`
+	Arch        string `json:"arch" toml:"arch"`
+	ArchVariant string `json:"arch-variant" toml:"arch-variant"`
+
+	Distribution *OSDistribution `json:"distribution,omitempty" toml:"distribution,omitempty"`
+}
+
+func (t *TargetMetadata) String() string {
+	var distName, distVersion string
+	if t.Distribution != nil {
+		distName = t.Distribution.Name
+		distVersion = t.Distribution.Version
+	}
+	return fmt.Sprintf("OS: %s, Arch: %s, ArchVariant: %s, Distribution: (Name: %s, Version: %s)", t.OS, t.Arch, t.ArchVariant, distName, distVersion)
+}
+
+type OSDistribution struct {
+	Name    string `json:"name" toml:"name"`
+	Version string `json:"version" toml:"version"`
+}
+
+// FIXME: the target logic in this file might be better located in a dedicated package.
```

Comment:
I kinda feel like this package should just be about serializing & deserializing data, leaving other CNB business logic to other packages. But I didn't want to change too much in one PR.

Reply:
I managed to move this stuff back to the platform package

Reply:
I managed to move this stuff back to the platform package